pkg/secrethub/internals/http/pagination.go

```diff
@@ -0,0 +1,41 @@
+package http
+
+import (
+	"net/url"
+)
+
+func NewPaginator(url url.URL, target interface{}, getStartingAfter func(interface{}) string, toItems func(interface{}) []interface{}, client *Client) *Paginator {
+	return &Paginator{
+		url:              &url,
+		target:           target,
+		getStartingAfter: getStartingAfter,
```

Comment:
Do we also support `getStartingBefore` or `getEndingBefore`? If so, how would that fit?

Reply:
No, that is not yet supported with this PR.

When we add functionality for that, I imagine doing so by injecting the next page behavior (strategy pattern) in the paginator, but we're flexible to choose whatever we want when we get to it. We can safely change this constructor, as it is only used internally (hence the `internals` package).

For exposing this functionality trough the `ListEvents` functions to the user, see: https://github.com/secrethub/secrethub-go/pull/144/files/83e130dc31279abcb00411bf94e53acb412c5f2f#r338147963

Reply:
I think you misunderstood my intention. I'm asking: does the API have this functionality and therefore should we add that functionality to the client as well?

Reply:
Yes, the API has this functionality. Yes, let's add it to the client as well. Let's do that in a separate PR though, to keep changes clear and concise and be able to release functionality sooner.